### PR TITLE
Upgrade Thanos and Prometheus

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
In this PR I'm upgrading Thanos and Prometheus, in order to introduced the config option for the TSDB head chunks write buffer size. In details:

1. Added `-blocks-storage.tsdb.head-chunks-write-buffer-size-bytes`
2. Mapped 2 new alertmanager metrics
3. Renamed `MetricMetadata` field `MetricName` to `MetricFamilyName` to match naming in Prometheus (in this PR it has been added to Prometheus too 🎉 )

I manually checked all Thanos changes and LGTM. There are no new Thanos metrics to map.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/client/cortex.proto [hunk 1]
```diff
@@ -165,7 +165,7 @@ message MetricMetadata {
   }
 
   MetricType type = 1;
-  string metric_name = 2;
+  string metric_family_name = 2;
```
[the remark this answers is not in the document]
Comment (gouthamve) [reply pkg/ingester/client/cortex.proto:168]: Also, why are we making this change? does it break anything?

**Review by gouthamve — APPROVED** [review]

File: pkg/ingester/client/cortex.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (pstibrany) [reply pkg/ingester/client/cortex.proto:168]: Field name changes don't break compatibility, as long as tag values are preserved.

**Review by pstibrany — APPROVED** [review]
Cortex changes lgtm. Quick look at Thanos changes also doesn't raise concerns. Haven't verified Prometheus and other libs changes.

File: pkg/ingester/client/cortex.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (pracucci) [reply pkg/ingester/client/cortex.proto:168]: > Does this break backwards compatibility?

No, field name change doesn't break it.

> why are we making this change?

To keep the same exact naming of Prometheus, given the same struct is defined in Prometheus too now.

**Pull request merged**